### ISSUE TITLE
Add QEMU Guest Agent package

### ISSUE THREAD

**Issue by ccesario** [opnsense/plugins#1586]
As already discussed in this thread https://forum.opnsense.org/index.php?topic=5860.15 It would be  nice to add this packto OPNSense ports.

https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=232949

**Comment by ccesario** [comment]
Hello, 
Is there any ETA for this port/plugin to be added!?

regards

**Comment by fichtner** [comment]
As said elsewhere, https://github.com/aborche/qemu-guest-agent is *not* in FreeBSD ports.

**Comment by ccesario** [comment]
@fichtner  Well, according this https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=232949  
I think that this is commited on FreeBSD ports

Regards,

**Comment by fichtner** [comment]
I am not sure how to convince you. I've laid out our limits to this request.

Let's try something else: what is the next step you expect from OPNsense? With this I mean something that can be committed to get one step further to "inclusion".

**Comment by AdSchellevis** [comment]
This issue has been automatically timed-out (after 180 days of inactivity).

For more information about the policies for this repository,
please read https://github.com/opnsense/plugins/blob/master/CONTRIBUTING.md for further details.

If someone wants to step up and work on this issue,
just let us know, so we can reopen the issue and assign an owner to it.

**Comment by ccesario** [comment]
Hey Guys, there is no chance to add this port to OPNSense?

**Comment by ccesario** [comment]
Without this package is not possible get Guest OS details In RHEV/Ovirt/OLVM Virtualization 

![image](https://user-images.githubusercontent.com/1812609/82256356-95429180-992c-11ea-9e6d-85c3a22e3dba.png)

Information like IP address / FQDN and other infos  are not supported, and the exclamation ICON is permanent on the VM

**Comment by fraenki** [comment]
> Hey Guys, there is no chance to add this port to OPNSense?

Nothing has changed. There is still no port for Qemu guest agent in FreeBSD's ports tree:
https://github.com/aborche/qemu-guest-agent/issues/8

I run OPNsense on oVirt too, and to be honest there is no _need_ to have guest tools installed. It's nice and all, but besides that there is no real benefit.

**Comment by fhloston** [comment]
Actually there is, backup of the vm with "guest-fsfreeze-freeze" would be clean instead of broken. I have corrupt image backups of opnsense installations on a regular basis.

**Comment by fichtner** [comment]
It sounds a bit like the problem described in https://github.com/opnsense/core/issues/4256 -- if the problem is runtime suspend+resume the answer is probably "don't do it". We neither write file system nor hypervisor code so that is our easy to follow recommendation to avoid integrity / operational damage.

**Comment by fhloston** [comment]
It is a ceph based rbd snapshot of the opnsense disk while opnsense is running. When I restore this backup the opnsense installation is often broken, reporting many disk corruptions.

This also happens if the VM with opnsense is hard-reset or turned off hard. Not everytime, but quite often.

Last issue also exists on lvm lv backed disks or native installations - so it is not ceph related.

But quite OT here...

**Comment by fichtner** [comment]
Let me rephrase: if snapshots of running VMs are not working use snapshots of properly shut down VMs

**Comment by toxic0berliner** [comment]
+1
I would also be interested in having qemu-guest-agent.
I would indeed be usefull even if fichtner doubts it ;)

**Comment by fichtner** [comment]
As soon as there is a package in FreeBSD I'll add it. Nothing has change here with that offer.

**Comment by revilo951** [comment]
Looks like this might be a promising step forward?

https://www.reddit.com/r/freebsd/comments/eehq9i/qemuguestagent_for_freebsd/

https://github.com/aborche/qemu-guest-agent

Tested on FreeBSD 12.

Sorry I don't know enough to be able to help with any of the real work on this, and thanks to those that do/can!

**Comment by fichtner** [comment]
Nobody pushed it to FreeBSD unfortunately...

% git grep bin.qemu-ga 
emulators/qemu-devel/pkg-plist:bin/qemu-ga
emulators/qemu/pkg-plist:bin/qemu-ga
emulators/qemu31/pkg-plist:bin/qemu-ga
emulators/qemu41/pkg-plist:bin/qemu-ga
emulators/qemu42/pkg-plist:bin/qemu-ga
emulators/qemu50/pkg-plist:bin/qemu-ga

Yes, there is a binary, but we will not add a **full** QEMU port because we would like to use the precious build time for more relevant binary packages.

It's easy enough to build it though. Instructions here https://docs.opnsense.org/manual/software_included.html?highlight=opnsense%20code%20ports#the-ports-tree


Cheers,
Franco

**Comment by fichtner** [comment]
PS: https://github.com/aborche/qemu-guest-agent/issues/8#issuecomment-633003557

**Comment by fraenki** [comment]
The qemu guest agent port finally landed in FreeBSD:
https://svnweb.freebsd.org/ports?view=revision&revision=568621
https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=254105

**Comment by fichtner** [comment]
@fraenki I promised to take it when that happens but if you want it I am happy to assist

**Comment by ccesario** [comment]
@fraenki Nice to know!! 👍

**Comment by fraenki** [comment]
@fichtner Enjoy your time off. :) I'll add a tiny GUI with a few options, which can later be extended as needed. (The qemu guest agent will work without touching the GUI, of course.)

**Comment by fraenki** [comment]
A preview/development version of os-qemu-guest-agent will probably be available in the next release of OPNsense. If no problems are reported, then the final version should follow shortly after.
